src/gengowatcher/secure_storage.py

```
@@ -0,0 +1,135 @@
+            encrypted_data = self._cipher.encrypt(json_data.encode())
+            
+            # Create directory if it doesn't exist
+            self.storage_file.parent.mkdir(parents=True, exist_ok=True)
+            
+            with open(self.storage_file, 'wb') as f:
+                f.write(encrypted_data)
+            
+            # Set restrictive file permissions (read/write for owner only)
+            try:
+                self.storage_file.chmod(0o600)
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)